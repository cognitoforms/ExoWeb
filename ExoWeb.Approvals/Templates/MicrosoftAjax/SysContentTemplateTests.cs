using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace ExoWeb.Approvals.Templates.MicrosoftAjax
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class SysContentTemplateTests : BaseRenderTests
	{
		private static readonly string templatesMarkup = @"
			<div class=""sys-template readonly"" sys:attach=""template"" template:name=""readonly"" template:datatype=""Request"">
				<span>{binding Description}</span>
			</div>
			<div class=""sys-template"" sys:attach=""template"" template:datatype=""Request"">
				<span>Description: <input type=""text"" sys:value=""{binding Description}"" /></span>
			</div>";

		protected override string Render(string pageMarkup)
		{
			return base.Render(templatesMarkup, pageMarkup);
		}

		[TestMethod]
		public void SysContentTemplate_Literal()
		{
			var outputMarkup = Render(@"
				<div class=""sys-template"" sys:attach=""dataview"" dataview:data=""{~ context.model.request, source=window }"" sys:content-template=""readonly"">
					<div sys:attach=""content"" content:data=""{{ $dataItem }}""></div>
				</div>");

			ApprovalTests.Approvals.VerifyHtml(outputMarkup);
		}

		[TestMethod]
		public void SysContentTemplate_CodeExpression()
		{
			var outputMarkup = Render(@"
				<div class=""sys-template"" sys:attach=""dataview"" dataview:data=""{~ context.model.request, source=window }"">
					<div sys:content-template=""{{ $dataItem.get_User().get_IsActive() ? '' : 'readonly' }}"" sys:attach=""content"" content:data=""{{ $dataItem }}""></div>
				</div>");

			ApprovalTests.Approvals.VerifyHtml(outputMarkup);
		}
	}
}
