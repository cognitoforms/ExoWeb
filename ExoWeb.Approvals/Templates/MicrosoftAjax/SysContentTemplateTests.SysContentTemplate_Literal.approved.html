<output xmlns:sys="javascript:Sys" xmlns:dataview="javascript:Sys.UI.DataView" xmlns:content="javascript:ExoWeb.UI.Content" xmlns:template="javascript:ExoWeb.UI.Template" xmlns:toggle="javascript:ExoWeb.UI.Toggle" xmlns:behavior="javascript:ExoWeb.UI.Behavior" xmlns:html="javascript:ExoWeb.UI.Html" xmlns:togglegroup="javascript:ExoWeb.UI.ToggleGroup">
  <div class="sys-template" id="exo0">
    <div sys:attach="content" content:data="{{ $dataItem }}"></div>
  </div>
  <div data-sys-attach="dataview" data-dataview-data="{~ context.model.request, source=window }" data-sys-content-template="readonly" data-sys-tmplidx="0" id="exo1" class="sys-ignore">
    <!--item:s1-->
    <div data-sys-attach="content" data-content-data="{{ $dataItem }}" data-sys-tcindex="s2" class="readonly">
      <span data-sys-innerhtml="{binding Description}">Something doesn't work</span>
    </div>
    <!--/item-->
  </div>
  <script type="text/javascript">$exoweb({ domReady: function() { Sys.Application.linkElement(document.getElementById("exo1"), document.getElementById("exo0")); } });</script>
</output>